src/lib/arch/riscv/InstructionMetadata.hh

```diff
@@ -52,30 +53,40 @@ struct InstructionMetadata {
 
   /** The instruction's mnemonic. */
   char mnemonic[CS_MNEMONIC_SIZE];
+
   /** The remainder of the instruction's assembly representation. */
   std::string operandStr;
 
   /** The implicitly referenced registers. */
   uint16_t implicitSources[MAX_IMPLICIT_SOURCES];
+
   /** The number of implicitly referenced registers. */
   uint8_t implicitSourceCount;
 
   /** The implicitly referenced destination registers. */
   uint16_t implicitDestinations[MAX_IMPLICIT_DESTINATIONS];
+
   /** The number of implicitly referenced destination registers. */
   uint8_t implicitDestinationCount;
 
   /** The explicit operands. */
   cs_riscv_op operands[MAX_OPERANDS];
+
   /** The number of explicit operands. */
   uint8_t operandCount;
 
+  /** The current exception state of this instruction. */
+  InstructionException metadataException_ = InstructionException::None;
+
+  /** Whether an exception has been encountered. */
+  bool metadataExceptionEncountered_ = false;
+
```

Comment:
These should be private with appropriate getters given they should only be modified by `InstructionMetadata.cc`